eth-contracts/contracts/Governance.sol

```diff
@@ -980,11 +1060,26 @@ contract Governance is InitializableV2 {
             "Governance: Must provide valid non-zero _proposalId"
         );
 
+        ServiceProviderFactory spFactory = ServiceProviderFactory(serviceProviderFactoryAddress);
+        DelegateManager delegateManager = DelegateManager(delegateManagerAddress);
+
+        // Amount directly staked by voter in service provider factory
+        (uint256 activeDeployerStake,,,,,) = spFactory.getServiceProviderDetails(_voter);
+        // Amount delegated by voter to other protocol participants in DelegateManager
+        uint256 activeDelegatorStake = delegateManager.getTotalDelegatorStake(_voter);
+        // Amount locked up in service provider factory
+        (uint256 lockedDeployerStake,) = spFactory.getPendingDecreaseStakeRequest(_voter);
+        // Amount locked up from voter address in DelegateManager
+        (,uint256 lockedDelegatorStake, ) = delegateManager.getPendingUndelegateRequest(_voter);
+
+        // Calculate total in ServiceProviderFactory as staked - locked
+        activeDeployerStake = activeDeployerStake.sub(lockedDeployerStake);
```

Comment:
test voting from delegator only account, confirm whether delegators should be able to submit proposals